narwhal/worker/src/worker.rs

```diff
@@ -42,6 +42,9 @@ pub mod worker_tests;
 /// The default channel capacity for each channel of the worker.
 pub const CHANNEL_CAPACITY: usize = 1_000;
 
+/// The maximum allowed size of transactions into Narwhal.
+pub const MAX_ALLOWED_TRANSACTION_SIZE: usize = 6 * 1024 * 1024;
```

Comment:
Thanks so much @mwtian  for putting this. I would recommend extracting this in next iteration to a config property so we (and our customers) tune it accordingly

Reply:
Yeah I can see projects integrating with Narwhal would want to tune this. But for Sui, this property (and GC depth, and a few other properties) should be made clear that validators should not change the value. Let me think about a good way to set this up.

Reply:
Perhaps having some structure such as `ProtocolConfig` and pass it from sui to narwhal for all such constants?

Reply:
Yes Sui needs to override this config and print a warning if it is set.

Reply:
Sorry, what I mean is that things that should be part of the "protocol configuration"(e.g. validators can not change) can be a part of a separate data structure.
E.g. right now we have narwhal_config::Parameters which is something that node operator can set.

As you mentioned things like `MAX_ALLOWED_TRANSACTION_SIZE` as well as some other fields in this structure are in fact protocol constants and there is no point to have node operators be able to change it. On the other side, some other users of narwhal protocol might want to define different numbers there.

So the solution I propose is that we have a separate structure (ProtocolConfig) that allows to parametrize things like this txn size limit and pass it to narwhal from sui when initialized. Sui will initialize this structure with some constants, and other users could put other numbers there (we can also just provide default impl)

Reply:
Yeah that is what I mean as well.

Reply:
@mwtian @andll good point on the "protocol configuration". In the past we discussed with @asonnino the possibility of moving such sensitive protocol properties (ex the `gc_depth` , `max_header_delay` etc) out side of any property file/struct and make it part of the `committee` it self - which gets updated during a reconfiguration as well when we transition to a new epoch and is basically a "product" of general acceptance of a new status quo . Seeing this discussion here I believe it worth continue it and decide what path we would like to take.